flink-scala/src/main/scala/org/apache/flink/api/scala/utils/package.scala

```diff
@@ -103,6 +105,25 @@ package object utils {
       : DataSet[T] = {
       wrap(jutils.sampleWithSize(self.javaSet, withReplacement, numSamples, seed))
     }
+
+    // --------------------------------------------------------------------------------------------
+    //  Checksum
+    // --------------------------------------------------------------------------------------------
+
+    /**
+      * Convenience method to get the count (number of elements) of a DataSet
+      * as well as the checksum (sum over element hashes).
+      *
+      * @return A ChecksumHashCode with the count and checksum of elements in the data set.
+      *
+      * @see [[org.apache.flink.api.java.Utils.ChecksumHashCodeHelper]]
+      */
+    def checksumHashCode: ChecksumHashCode = {
```

Comment:
Would be good to give this method parenthesis. It triggers distributed execution, so is not quite side-effect free.